src/app/dataset/datasetform/datasetform.component.ts

```diff
@@ -85,12 +85,45 @@ export class DatasetformComponent implements OnInit {
     this.buildForm();
     this.returnCountries();
     this.returnLanguages();
-
     this.invalidNotification = errorNotification(this.translate.instant('formerror'), {
       sticky: true
     });
   }
 
+  /** redirectionIds
+  /* - returns form control as array
+  /* - (necessary for template-check to pass)
+  */
+  get redirectionIds(): FormArray {
+    return this.datasetForm.get('redirectionIds') as FormArray;
+  }
+
+  /** addRedirectionId
+  /* - handle addition to the redirectionIds array
+  /* @param {string} val - the item to add
+  */
+  addRedirectionId(val: string): void {
+    const ids = this.redirectionIds;
+    const existingIndex = ids.value.findIndex((id: string) => id === val);
+    if (existingIndex === -1) {
+      ids.push(this.fb.control(val));
+      this.datasetForm.markAsDirty();
+    }
+  }
+
+  /** removeRedirectionId
+  /* - handle removal from the redirectionIds array
+  /* @param {string} m - the item to remove
+  */
+  removeRedirectionId(val: string): void {
+    const ids = this.redirectionIds;
+    const existingIndex = ids.value.findIndex((id: string) => id === val);
+    if (existingIndex > -1) {
+      ids.removeAt(existingIndex);
+      this.datasetForm.markAsDirty();
+    }
+  }
+
```

Comment:
As these two methods are almost identical, can you somehow combine the code? Eg, where you pass in if it should be removed or added?

Reply:
Done